core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -141,8 +141,8 @@ public CallOptions withRequestKey(@Nullable RequestKey requestKey) {
    */
   @ExperimentalApi
   @Nullable
-  public RequestKey getRequestKey() {
-    return requestKey;
+  public Attributes getAttributes() {
```

Comment:
I would much rather have one API for setting attributes.  Having both withAttributes() as whole-sale replacement and withAttribute() for updates may be confusing to some as to the expected behavior.  For updating attributes I think it makes more sense to have an API like,

```java
Attribute orig = callOptions.getAttributes();
Attribute toAdd = Attributes.newBuilder().set(key, value);
callOptions.withAttributes(orig.add(toAdd));
```

Regarding the method name, how about just calling it withAffinity, 
```java
CallOptions#withAffinity(Attributes attr);
```